### ISSUE TITLE
Why is cancel reason not part of onJobCancelled callback?

### ISSUE THREAD

**Issue by ZakTaccardi** [yigit/android-priority-jobqueue#286]
I'm creating an RxWrapper around this library, and I'm observing all jobs via setting a callback. I'm using `singleInstanceId` to avoid duplicate jobs. 

An ideal API for my use case is that I would like to be able to observe the current state (lifecycle) of any job by its `singleInstanceId`.

```
enum class JobLifecycle {
    /**
     * Pending. Not yet running
     */
    WILL_RUN,
    /**
     * Currently running
     */
    RUNNING,
    /**
     * Failed, but will try again
     */
    WILL_RETRY,
    /**
     * Failed, will not be run again
     */
    FAIL,
    /**
     * Success!
     */
    SUCCESS
}
```

Two problems I'm having:

1. Duplicate jobs: 

- I have two jobs of the same class, `jobInstanceA` and `jobInstanceB`
- both have the same `singleInstanceId`. 
- `jobInstanceA#onJobAdded()` (fires a `WILL_RUN` event)
- `jobInstanceA#onJobRun()` (fires a `RUNNING` event)
- `jobInstanceB#onJobAdded` (fires a `WILL_RUN` event, overwriting the previous `RUNNING` event)`. Because `jobInstanceB#onJobAdded` is a duplicate, I don't care at all about this job. I don't want this `WILL_RUN` event to overwrite the correct `RUNNING` event for this job's `singleInstanceId`. Is there a way to check if a job is a duplicate in the `onJobAdded` callback?

2. No cancel reason in `onJobCancelled`
```
    void onJobCancelled(@NonNull Job job, boolean byCancelRequest, @Nullable Throwable throwable);
```
- I want to ignore cancel events for jobs that are duplicates. How can I do that, if the `@CancelReason int cancelReason` is not available in this callback?

**Comment by yigit** [comment]
makes sense to make the requested API change. Meanwhile, you can get the tags of the job to do the matching.
